pandas/indexes/multi.py

```diff
@@ -66,8 +66,10 @@ def __new__(cls, levels=None, labels=None, sortorder=None, names=None,
                 name=None, **kwargs):
 
         # compat with Index
-        if name is not None:
-            names = name
+        if name is None and hasattr(levels, 'name'):
+            name = levels.name
+        if isinstance(levels, MultiIndex):
+            return levels.copy(name=name, names=names, deep=copy)
```

Comment:
*this* could also be true I agree, but then it should be *before* the prior check, yes?
I also really don't like the ``name``, ``names`` biz. I believe ``Index`` has the right way to do this (IOW check for both ``name`` and ``names`` if both are specified its an error, otherwise use the non ``None`` one)